### DISCUSSION TITLE
Mixed Page Orientations in a Single Document

### DISCUSSION THREAD

**Discussion by SkysmithGit** [QuestPDF/QuestPDF#511] [Q&A] [answered]
I am trying to create a PDF document with a collection of image files. I want to change the page orientation (landscape vs portrait) per page, in accordance with the image on the page.

```
            Document.Create(container =>
            {
                var files = LV_Files.Items.Cast<object>().ToArray();
                foreach (string file in files.Cast<string>())
                {
                    System.Drawing.Image image = System.Drawing.Image.FromFile(file);

                    container
                        .Page(page =>
                        {
                            if (image.Width > image.Height)
                            {
                                page.Size(PageSizes.Letter.Landscape());
                            }
                            else
                            {
                                page.Size(PageSizes.Letter.Portrait());
                            }

                            page.Content()
                                .AlignMiddle()
                                .AlignCenter()
                                .Column(x =>
                                {
                                    x.Item().Image($"{file}", ImageScaling.FitArea);
                                });
                        });
                }
            }).GeneratePdf("images.pdf");

```
Somehow, the page orientation is only set once for the first page and other pages just follow the first page.

How do I fix this?

**Comment by SkysmithGit** [comment, marked as answer]
I was playing with the code a bit more and I think it is related to the photo's metadata.

The camera I used to take the photos automatically determines if the photo is horizontal or vertical. When the photo is added to a page, it displays correctly, but somehow it is still considered as horizontal. Regardless its actual orientation.

I tried the code with vertical images from different sources and the code works fine.

**Reply by girlpunk** [reply]
You may be encountering a "feature" of JPEG images, wherein internal metadata is used to instruct the viewer to rotate the image before displaying instead of storing the image with the correct rotation.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
